plugins/scaffolder/src/components/fields/EntityTagsPicker/EntityTagsPicker.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useState } from 'react';
+import { useAsync, useEffectOnce } from 'react-use';
+import { CatalogEntitiesRequest } from '@backstage/catalog-client';
+import { Entity, makeValidator } from '@backstage/catalog-model';
+import { useApi } from '@backstage/core-plugin-api';
+import { catalogApiRef } from '@backstage/plugin-catalog-react';
+import { FormControl, TextField } from '@material-ui/core';
+import { Autocomplete } from '@material-ui/lab';
+import { FieldProps } from '@rjsf/core';
+
+export const EntityTagsPicker = ({
+  formData,
+  onChange,
+  uiSchema,
+}: FieldProps<string[]>) => {
+  const catalogApi = useApi(catalogApiRef);
+  const [inputValue, setInputValue] = useState('');
+  const [inputError, setInputError] = useState(false);
+  const tagValidator = makeValidator().isValidTag;
+  const kinds = uiSchema['ui:options']?.kinds as string[];
+
+  const { loading, value: existingTags } = useAsync(async () => {
+    const tagsRequest: CatalogEntitiesRequest = { fields: ['metadata.tags'] };
+    if (kinds) {
+      tagsRequest.filter = { kind: kinds };
```

Comment:
Hold up, this is the `and` filter right? meaning that if you list multiple kinds then an entity has to match all of those kinds at once

Reply:
https://github.com/backstage/backstage/blob/af4980fb5d304fe0ec49f85fbe89b5c5e4c83d63/packages/catalog-client/src/types/api.ts#L43-L60

Also confirmed the behavior as stated above. `EntityPicker` also does the same thing :)

https://github.com/backstage/backstage/blob/af4980fb5d304fe0ec49f85fbe89b5c5e4c83d63/plugins/scaffolder/src/components/fields/EntityPicker/EntityPicker.tsx#L46

Reply:
.__________.

So many levels. I mean it is nicer than having to `kinds.map(kind => ({ kind }))` I guess? 😅